### ISSUE TITLE
Swapping Buttons Messes Up The Order Of Other Buttons

### ISSUE THREAD

**Issue by martinemmanuelsantos** [rivafarabi/deckboard#372]
When I want to swap the position of two buttons, it reorders all the buttons around it and makes it very difficult to organize pages. See attached video.

https://streamable.com/bt154

**Comment by Yurlyn** [comment]
I believe that's intended. You are the one swapping the buttons by moving a button back to an earlier spot so all the other buttons will shift in order to the empty spot. This is also to prevent overlapping buttons and since you're moving the button onto a button slot that's in use this is better to make sure you don't actually get that overlap or even erase a button. Maybe Riva could add a feature so when a user drags a button to an empty slot though it won't shift everything and when dragging an empty slot onto another empty slot or button slot it will shift. 

Anyway, you can work around this by swapping empty buttons around so it keeps your intended order in the end.

**Comment by martinemmanuelsantos** [comment]
Thanks for getting back to me. I'm aware you can do that, but this is really inconvenient for large decks. I think it would work better if it worked like the Stream Deck software. e.g. The dragged button should *swap* with the button that you drop it on. This would prevent all the buttons from shifting around.

**Comment by mix3d** [comment]
I agree that the mentioned expected behavior should be default. I've run into the same issue reordering my buttons. You could probably add support for both interactions; if the drag position is within 50% of the center of a button, swap, otherwise if it's between two buttons, move other buttons to make space for the moved one.
